adapters/unruly/unruly_test.go

```diff
@@ -0,0 +1,245 @@
+package unruly
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/adapters/adapterstest"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestJsonSamples(t *testing.T) {
+	adapterConfig := new(adapters.HTTPAdapterConfig)
+	adapterstest.RunJSONBidderTest(t, "unrulytest", NewUnrulyAdapter(adapterConfig, "http://targeting.unrulymedia.com/openrtb/2.2"))
```

Comment:
This will run tests defined in JSON that can test a wide range of issues. See all the JSON files in other adapters in `<adapterName>test` directories of the adapters. You may be able to convert many of your tests below into JSON format, and perhaps make them even more comprehensive. The JSON also helps other people understand what is and is not supposed to work as a request for your adapter.